test/com/facebook/buck/apple/AppleBundleIntegrationTest.java

```diff
@@ -632,6 +632,16 @@ public void appBundleWithConflictingResources() throws Exception {
     workspace.runBuckCommand("build", target.getFullyQualifiedName()).assertFailure();
   }
 
+  @Test
+  public void appBundleWithConflictingNestedFolderResources() throws Exception {
+    ProjectWorkspace workspace =
+        TestDataHelper.createProjectWorkspaceForScenario(this, "app_bundle_with_conflicting_nested_resources", tmp);
+    workspace.setUp();
+
+    BuildTarget target = workspace.newBuildTarget("//:DemoApp#iphonesimulator-x86_64,no-debug");
+    workspace.runBuckCommand("build", target.getFullyQualifiedName()).assertFailure();
```

Comment:
@rmaz We should add another test that actually tests for success but has files with the same name inside the folders. We can just copy `app_bundle_with_conflicting_nested_resources` and change:

    files = glob(["**/*.png"]),

to:

    dirs = ["ImagesA", "ImagesB"],

Alternatively, we can just skip the images as the localizable strings all have the same file names inside.